### PR TITLE
[#9206] fix wrong ping handling on signing messages

### LINKED ISSUE #9206: Endless processing spinner after entering wrong PIN when signing message from DApp
# Bug Report

## Problem

If you enter wrong PIN instead of validation error endless "Processing" spinner is shown.
Also translation is missing on signing screen.
miss

#### Expected behavior
validation error, can enter valid PIN

#### Actual behavior
proc_s

#### Notes

For the developer who will be investigating. Is it related to another PR? Tips on where to start? 

#### Acceptance Criteria

Rules for the future PR to be accepted.

### Reproduction

- Open Status
- Login with KeyCard
- Open status-im.github.io/dapp > Transaction > Sign message
- Enter wrong PIN-code > sign in

### Additional Information

[comment]: # (Please do your best to fill this out.)
- Status version: nightly 14/11/2019
- Operating System: Android

#### Logs

Full logcat: [logcat.log](https://github.com/status-im/status-react/files/3725121/resend_button.log)

### PR DESCRIPTION
**Pull request by rasom**
fix #9206 

status: ready

### CODE REVIEW

File: src/status_im/hardwallet/core.cljs [hunk 1]
```diff
@@ -1843,19 +1843,21 @@
   (let [tag-was-lost? (= "Tag was lost." (:error error))]
     (fx/merge cofx
               (if tag-was-lost?
```
Comment (yenda) [inline src/status_im/hardwallet/core.cljs]: isn't it a when?

File: src/status_im/hardwallet/core.cljs [hunk 2]
```diff
@@ -1843,19 +1843,21 @@
...
-                          (get-application-info (get-pairing db) nil))
+                (fn [{:keys [db] :as cofx}]
+                  (fx/merge cofx
+                            {:db (-> db
+                                     (update-in [:hardwallet :pin] merge {:status      :error
+                                                                          :sign        []
+                                                                          :error-label :t/pin-mismatch})
+                                     (assoc-in [:signing/sign :keycard-step] :pin))}
+                            (get-application-info (get-pairing db) nil)))
                 (show-wrong-keycard-alert cofx true)))))
```
Comment (yenda) [inline src/status_im/hardwallet/core.cljs]: isn't this one still wrong? it looks like it is part of the fx/merge and is passed an extra cofx?

File: src/status_im/hardwallet/core.cljs (hunk 1 above)
Comment (rasom) [reply src/status_im/hardwallet/core.cljs]: doesn't really change here anything and I just didn't touch it :)

File: src/status_im/hardwallet/core.cljs (hunk 2 above)
Comment (rasom) [reply src/status_im/hardwallet/core.cljs]: Yeah it's broken (+ in one more place in the same file which wasn't shown in PR). Although original issue is fixed even with this bug. I didn't notice that, will fix in this pr

Comment (rasom) [reply src/status_im/hardwallet/core.cljs]: fixed

File: src/status_im/hardwallet/core.cljs (hunk 1 above)
Comment (rasom) [reply src/status_im/hardwallet/core.cljs]: fixed

**Review by yenda — APPROVED** [review]

**Pull request merged**